### PR TITLE
Makes models

### PR DESCRIPTION
**Pull request by Tungtt2020**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -9,36 +9,63 @@ info:
...
   - name: Make
-
+  - name: Model
 paths:
-  /v1/makes:
+  /public/v1/makes:
     get:
       operationId: GetMakes
-      description: Returns a list of makes sorted alphabetically.
+      description: Returns a sorted list of makes for a vehicle category
```
Comment (aleboj) [inline openapi.yaml:24]: missing `.` at the end of the sentence

File: openapi.yaml [hunk 2]
```diff
@@ -9,36 +9,63 @@ info:
...
-                type: array
-                items:
-                  "$ref": "#/components/schemas/Make"
-
+                $ref: "#/components/schemas/MakeResponse"
+  /public/v1/makes/key/{makeKey}/models:
+    get:
+      tags:
+        - Model
+      description: Returns a sorted list of models by makeKey and vehicleCategory
```
Comment (aleboj) [inline openapi.yaml:42]: missing `.` at the end of the sentence

File: openapi.yaml [hunk 3]
```diff
@@ -9,36 +9,63 @@ info:
...
+          in: path
+          required: true
+          schema:
+            type: string
+        - $ref: "#/components/parameters/vehicleCategoryQueryParam"
+        - $ref: "#/components/parameters/pageQueryParam"
+        - $ref: "#/components/parameters/sizeQueryParam"
+      responses:
+        "200":
+          description: Models for a makeKey and vehicle category returned
```
Comment (aleboj) [inline openapi.yaml:55]: missing `.` at the end of the sentence

File: openapi.yaml [hunk 4]
```diff
@@ -61,24 +88,113 @@ components:
...
+    Model:
+      description: Represents the data for the Model entity.
+      type: object
+      properties:
+        key:
+          description: Unique key of the model.
+          type: string
+          example: "a4-allroad"
+        name:
+          description: Display name of the model
```
Comment (aleboj) [inline openapi.yaml:124]: missing `.` at the end of the sentence

File: openapi.yaml [hunk 5]
```diff
@@ -61,24 +88,113 @@ components:
...
+        number:
+          description: Page number
+          type: integer
+          format: int32
+          example: 6
+        totalPages:
+          description: Total number of pages
+          type: integer
+          format: int32
+          example: 7
```
Comment (aleboj) [inline openapi.yaml:185]: missing `.` at the end of the sentence

File: openapi.yaml [hunk 6]
```diff
@@ -61,24 +88,113 @@ components:
         format: int32
         minimum: 1
         maximum: 2000
-        default: 25
+        default: 20
```
Comment (aleboj) [inline openapi.yaml:91]: isn't it 25?

Comment (Tungtt2020) [reply openapi.yaml:91]: no, it is 20 by default

**Review by milanovici — APPROVED** [review]
👍

**Review by lukicsasa — APPROVED** [review]
Looks good to me

**Comment by Tungtt2020** [comment]
i also aligned nested schema name to swagger validator, removed suffix dto, adjusted default page size example, extracted enum types to reusable objects.
checked with swagger UI and got valid badge.
image
image

**Pull request merged**